sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -161,7 +161,11 @@ object FileSourceStrategy extends Strategy with PredicateHelper with Logging {
       val filterSet = ExpressionSet(filters)
```

Comment:
I believe I fixed all places that can use `PythonUDF` as partitioning filters. For DSv1, we don't pushdown partitioning filters. `PythonUDF` could be pushed down, and the downstream datasource can still handle them if they want.

So I fixed:
- Hive scan with partitoning
- DSv2 with partitioning
- FileSource with partitioning